facet-toml/src/to_scalar.rs

```diff
@@ -142,9 +155,12 @@ pub(crate) fn put_from_str<'input, 'a>(
                 got: item.type_name(),
             },
             item.span(),
+            wip.path(),
         )
     })?;
 
+    // TODO: only generate if actually error
+    let path = wip.path();
```

Comment:
Is `wip.path()` meaningful to the user?

At least with serde, the Rust data model can deviate from the Serde data due to features like `untagged`, `flatten`.  What `toml_edit` does in its serde code is to add the table keys to the error as its unwinding the stack on errors.